M2/cmake/check-libraries.cmake

```diff
@@ -82,8 +82,11 @@ endif()
 if(WITH_FFI)
   pkg_get_variable(FFI_INCLUDE_DIR libffi includedir)
   find_package(FFI REQUIRED QUIET)
-  execute_process(COMMAND pkg-config --modversion libffi
-    OUTPUT_VARIABLE LIBFFI_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+  if(NOT DEFINED LIBFFI_VERSION)
+    execute_process(COMMAND pkg-config --modversion libffi
+      OUTPUT_VARIABLE LIBFFI_VERSION_ OUTPUT_STRIP_TRAILING_WHITESPACE)
+    set(LIBFFI_VERSION ${LIBFFI_VERSION_} CACHE STRING "FFI version")
+  endif()
```

Comment:
I think it would be much cleaner if this code block was inside `FindFFI.cmake`. FindModules are usually supposed to set the version variable anyway, so this would be an improvement to the current one.